File: 2536-increment-submatrices-by-one/2536-increment-submatrices-by-one.py
import numpy as np
class Solution:
    def rangeAddQueries(self, n: int, queries: List[List[int]]) -> List[List[int]]:
        mat = [[0]*n for _ in range(n)]
        # mat = np.array(mat)
        # for a,b,c,d in queries:
            # mat[a:c+1, b:d+1] += 1
        
        for x1,y1,x2,y2 in queries:
            
            for i in range(x1,x2+1):
                mat[i][y1]+=1
                if y2+1<n:
                    mat[i][y2+1]-=1
        for i in range(n):
            v=0
            for j in range(n):
                v+=mat[i][j]
                mat[i][j]=v
        return mat

File: 2536-increment-submatrices-by-one/2536-increment-submatrices-by-one.java
class Solution {
    public int[][] rangeAddQueries(int n, int[][] queries) {
        int[][] mat = new int[n][n];
        for (int[] query : queries) {
            int row1 = query[0], col1 = query[1], row2 = query[2], col2 = query[3];
            for (int i = row1; i <= row2; i++) {
                for (int j = col1; j <= col2; j++) {
                    mat[i][j]++;
                }
            }
        }
        return mat;
    }
}